### ISSUE TITLE
UserWarning regarding the RSquare metric when used in Keras model 

### ISSUE THREAD

**Issue by philipwang97** [tensorflow/addons#2528]
\Anaconda3\lib\site-packages\tensorflow\python\keras\metrics.py:254: UserWarning: Metric RSquare implements a `reset_states()` method; rename it to `reset_state()` (without the final "s"). The name `reset_states()` has been deprecated to improve API consistency.
  warnings.warn('Metric %s implements a `reset_states()` method; rename it '

Tensorflow 2.5.0
Keras 2.5.0
tensorflow_addos 0.13.0
